Configuration.cpp

```diff
@@ -60,6 +60,33 @@ bool Configuration::readConfigFile(char *path) {
 
     default_marker = (m == "PASS") ? MARKER_PASS : MARKER_DROP;
   }
+  
+  //Possibile aggiungere più stati
+
+  if(!root["continents"].empty()) {
+    if(root["continents"]["whitelist"].empty()) {
+      //trace->traceEvent(TRACE_INFO, "Missing %s from %s", "whitelist", path);
+    } else {
+      for(Json::Value::ArrayIndex i = 0; i != root["countries"]["whitelist"].size(); i++) {
+	std::string continent = root["continents"]["whitelist"][i].asString();
+
+	trace->traceEvent(TRACE_INFO, "Adding %s to continents whitelist", continent.c_str());
+	continents[country2u16((char*)continent.c_str())] = MARKER_PASS;
+      }
+    }
+
+    if(root["continents"]["blacklist"].empty()) {
+      //trace->traceEvent(TRACE_INFO, "Missing %s from %s", "blacklist", path);
+    } else {
+      for(Json::Value::ArrayIndex i = 0; i != root["continents"]["blacklist"].size(); i++) {
+	std::string continent = root["continents"]["blacklist"][i].asString();
+
+	trace->traceEvent(TRACE_INFO, "Adding %s to continents blacklist", continent.c_str());
+	continents[country2u16((char*)continent.c_str())] = MARKER_DROP;
+      }
+    }
+  }
+
```

Comment:
Can we refactor the code in order to merge continent and country into the same parsing logic?

Reply:
We will resolve in the pull of Thursday